libfreerdp/codec/xcrush.c

```diff
@@ -734,11 +734,31 @@ static int xcrush_generate_output(XCRUSH_CONTEXT* xcrush, BYTE* OutputBuffer, UI
 
 static size_t xcrush_copy_bytes(BYTE* dst, const BYTE* src, size_t num)
 {
-	size_t index;
-
-	for (index = 0; index < num; index++)
+	if (src + num < dst || dst + num < src)
+	{
+		memcpy(dst, src, num);
+	}
+	else
 	{
-		dst[index] = src[index];
+		size_t index = 0;
+		// copy 4 or 8 bytes at once
+#if __LP64__
```

Comment:
Removed this, the compilers know how to handle vectorization, you just disabled this path for a lot of compilers/architectures

Reply:
I will use the last code I have shown